addon/-private/system/model/internal-model.js

```diff
@@ -889,13 +883,34 @@ export default class InternalModel {
     this.eachRelationship((name, relationship) => {
       if (this._relationships.has(name)) {
         let rel = this._relationships.get(name);
+
         rel.clear();
         rel.removeInverseRelationships();
       }
     });
     Object.keys(this._implicitRelationships).forEach((key) => {
-      this._implicitRelationships[key].clear();
-      this._implicitRelationships[key].removeInverseRelationships();
+      let rel = this._implicitRelationships[key];
+
+      rel.clear();
+      rel.removeInverseRelationships();
+    });
+  }
+
+  /*
+    @method unloadDeletedRecordFromRelationships
+   */
+  unloadDeletedRecordFromRelationships() {
```

Comment:
Also let's add some comments here as breadcrumbs to our future selves who will clean this up.

This does fix an important bug, but currently there are two codepaths whose different semantics are not obvious within the same level of abstraction, ie that `rel.clear()` causes a `flushCanonical` but `unloadDeletedInverseInternalModel ` does not is not clear, and that this is the entire point is also not clear.